test/WrappedMToken.t.sol

```diff
@@ -261,6 +487,57 @@ contract WrappedMTokenTests is Test {
         assertEq(_wrappedMToken.balanceOf(_alice), 1_100);
     }
 
+    function testFuzz_claimFor(uint240 balance_, uint128 accountIndex_, uint128 index_) external {
+        accountIndex_ = uint128(bound(index_, _EXP_SCALED_ONE, 10 * _EXP_SCALED_ONE));
+        balance_ = uint240(bound(balance_, 0, _getMaxAmount(accountIndex_)));
+        index_ = uint128(bound(index_, accountIndex_, 10 * _EXP_SCALED_ONE));
+
+        _registrar.setListContains(_EARNERS_LIST, address(_wrappedMToken), true);
+
+        _wrappedMToken.enableEarning();
+
+        _wrappedMToken.setTotalEarningSupply(balance_);
+
+        _wrappedMToken.setAccountOf(_alice, balance_, accountIndex_);
+
+        _mToken.setCurrentIndex(index_);
+
+        uint240 accruedYield_ = _wrappedMToken.accruedYieldOf(_alice);
+
+        assertEq(_wrappedMToken.claimFor(_alice), accruedYield_);
+
+        assertEq(_wrappedMToken.totalEarningSupply(), _wrappedMToken.balanceOf(_alice));
+    }
+
+    /* ============ claimExcess ============ */
+    function testFuzz_claimExcess(
+        uint128 index_,
+        uint240 totalNonEarningSupply_,
+        uint112 principalOfTotalEarningSupply_,
+        uint240 mBalance_
+    ) external {
+        index_ = uint128(bound(index_, _EXP_SCALED_ONE, 10 * _EXP_SCALED_ONE));
+
+        totalNonEarningSupply_ = uint240(bound(totalNonEarningSupply_, 0, _getMaxAmount(index_)));
+
+        uint240 totalEarningSupply_ = uint112(bound(principalOfTotalEarningSupply_, 0, _getMaxAmount(index_)));
+
+        principalOfTotalEarningSupply_ = uint112(totalEarningSupply_ / index_);
+
+        mBalance_ = uint240(bound(mBalance_, totalNonEarningSupply_ + totalEarningSupply_, type(uint240).max));
+
+        _mToken.setBalanceOf(address(_wrappedMToken), mBalance_);
+        _wrappedMToken.setTotalNonEarningSupply(totalNonEarningSupply_);
+        _wrappedMToken.setPrincipalOfTotalEarningSupply(principalOfTotalEarningSupply_);
+
+        _mToken.setCurrentIndex(index_);
+
+        uint240 excess = _wrappedMToken.claimExcess();
+
+        assertEq(_wrappedMToken.excess(), 0);
+        assertEq(_mToken.balanceOf(_wrappedMToken.vault()), excess); // TODO: Do with expectCall.
```

Comment:
Not sure what this comment implies.
Should we test that `IMTokenLike(mToken).transfer(vault, excess_);` is called?

Reply:
Yes. Not good practice to test state of Mocks.